storage.lua

```diff
@@ -43,23 +43,26 @@ function mail.get_message(playername, msg_id)
 	end
 end
 
-function mail.sort_messages(messages, sortfield, sortdirection)
-	local function sorter(field, dir)
-		return dir == "2"
-			and (function(a, b) return a[field] > b[field] end)
-			or (function(a, b) return a[field] < b[field] end)
-	end
-	local result = {unpack(messages)}
-	if sortfield == "1" then -- for inbox
-		table.sort(result, sorter("from", sortdirection))
-	elseif sortfield == "1" then -- for outbox
-		table.sort(result, sorter("to", sortdirection))
-	elseif sortfield == "2" then
-		table.sort(result, sorter("subject", sortdirection))
-	else -- default sorting, sortfield == "3"
-		table.sort(result, sorter("time", sortdirection))
+function mail.sort_messages(messages, sortfield, descending)
+	local results = {unpack(messages)}
+	if sortfield ~= nil then
+		if descending then
+			table.sort(results, function(a, b)
+				if a[sortfield] and b[sortfield] then
+					return a[sortfield] > b[sortfield]
+				end
+				minetest.log("warning", "mail.sort_messages: missing field "..sortfield)
+			end)
```

Comment:
@BuckarooBanzay as you've rewritten storage you might be able to answer if checks are necessary or not.
`sortfield` is assumed to always be one of the common fields found from mail table.

I mean one of these fields: https://github.com/mt-mods/mail/blob/da05b5d108d12c7487ffc2f97358ce5ef21835b7/api.lua#L73-L82

If, in normal operating conditions, messages should always contain all of those fields (anything not nil) then I think it would be better to remove these checks / logging and instead let it crash if it hits nil for some reason.

Second opinion maybe?

Reply:
Does not matter, removed checks. I guess better to let it crash if there's still missing fields and fix issue by making sure that all fields are available.